hackathon_site/dashboard/frontend/src/slices/hardware/hardwareSlice.ts

```diff
@@ -116,13 +116,21 @@ const hardwareSlice = createSlice({
             state.error = null;
         });
 
-        builder.addCase(getHardwareWithFilters.fulfilled, (state, { payload }) => {
-            state.isLoading = false;
-            state.error = null;
-            state.next = payload.next;
-            state.count = payload.count;
-            hardwareAdapter.setAll(state, payload.results);
-        });
+        builder.addCase(
+            getHardwareWithFilters.fulfilled,
+            (state, { payload, meta }) => {
+                state.isLoading = false;
+                state.error = null;
+                state.next = payload.next;
+                state.count = payload.count;
+
+                if (meta.arg?.keepOld) {
```

Comment:
When would we use keep old? refresh button?

Reply:
I use it on the cart page when some hardware is missing (which will happen if we eventually only persist the cart to local storage). Rather than replacing all the hardware in the store, I just fetch the ones that are missing.

Originally I wrote a separate thunk for it, but this was far easier.